src/ring_theory/finiteness.lean

```diff
@@ -90,7 +90,20 @@ variables (R)
 instance self : finite R R :=
 ⟨⟨{1}, by simpa only [finset.coe_singleton] using ideal.span_singleton_one⟩⟩
 
-variables {R}
+variable (M)
+
+lemma of_restrict_scalars_finite [module A M] [is_scalar_tower R A M] [hM : finite R M] :
+  finite A M :=
```

Comment:
I mean
```suggestion
  finite A (restrict_scalars R A M) :=
```

Reply:
This is weird, isn't it? `(restrict_scalars R A M)` morally is an `R`-module, not an `A`-module. In practice it works, since at the end of the day it's always `M`, but still...

Reply:
`restrict_scalars R A M` is '`M` with an _additional_ `R`-module structure', not "`M` with its `A`-module structure replaced with an `R`-module structure".

I think you're right though, and this is weird after all; in this context you're already assuming `module R M`, and `restrict_scalars R A M` is for when you only have `algebra R A` and `module A M` available. When you don't have `module R M` available you can't even state `finite R M` anyway.

Reply:
Maybe a more reasonable version is
```lean
lemma of_restrict_scalars_finite' (M : Type*) [add_comm_group M] [module A M]
  [hM : finite R ((restrict_scalars R A M))] : finite A M :=
by convert of_restrict_scalars_finite R A (restrict_scalars R A M)
```
(If I make this an instance I get the same error.)

Reply:
I was going to say that too, but you're right it's also ineligible for typeclass search, and it seems unlikely that it would every be used.